File: test/factories/accounts.rb
FactoryGirl.define do
	factory :account, aliases: [:bank_account] do
		account_type "bank"
		sequence(:name) { |n| "#{account_type} account #{n}" }
		opening_balance 1000
		status "open"

		ignore do
			num_transactions 0
		end

		after :build do |account, evaluator|
			create_list :basic_transaction, evaluator.num_transactions, account: account
		end

		trait :credit do
			account_type "credit"
		end

		trait :investment do
			account_type "investment"
			opening_balance 0
			after(:build) do |account|
				account.related_account = build(:cash_account)
				account.related_account.related_account = account
			end
		end

		trait :cash do
			account_type "cash"
		end

		trait :loan do
			account_type "loan"
		end

		trait :closed do
			status "closed"
		end

		factory :credit_account, traits: [:credit]
		factory :investment_account, traits: [:investment]
		factory :cash_account, traits: [:cash]
		factory :loan_account, traits: [:loan]
		factory :closed_account, traits: [:closed]
	end
end

File: spec/public/mocks/accounts/providers/accounts.js
(function() {
	"use strict";

	/**
	 * Registration
	 */
	angular
		.module("lootAccountsMocks")
		.provider("accountsMock", Provider);

	/**
	 * Implementation
	 */
	function Provider() {
		var provider = this;

		// Mock accounts object
		provider.accounts = [
			{id: 1, name: "aa", account_type: "bank", opening_balance: 100, status: "open"},
			{id: 2, name: "bb", account_type: "investment"},
			{id: 3, name: "cc", num_transactions: 1},
			{id: 4, name: "ba", account_type: "asset"},
			{id: 5, name: "ab", account_type: "asset"},
			{id: 6, name: "bc", account_type: "investment"},
			{id: 7, name: "ca"},
			{id: 8, name: "cb", account_type: "asset"},
			{id: 9, name: "ac"}
		];

		provider.$get = function() {
			// Return the mock accounts object
			return provider.accounts;
		};
	}
})();
